app/models/ht_user.rb

```diff
@@ -1,5 +1,34 @@
 # frozen_string_literal: true
 
 class HTUser < ApplicationRecord
-  # self.table_name = "ht_users"
+  # Brain-dead IPv4 validation
+  validates :iprestrict, presence: true,
+                         format: { with: /\A\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\z/,
```

Comment:
Check out https://www.oreilly.com/library/view/regular-expressions-cookbook/9780596802837/ch07s16.html for a battle-tested ip regex

Reply:
Battle-tested regex swapped in, except ^ and $ replaced by \A and \z to keep Ruby happy.

Reply:
I'd use `\Z` (ignores trailing newline) and maybe add some `\s*` on the front and back so we don't break on leading/trailing spaces.

Reply:
Comments distilled into an edit of HT-1494.
Added a strip() to deal with whitespace for the current PR.